README.md

```diff
@@ -1,8 +1,132 @@
 # tomland
 
-[![Hackage](https://img.shields.io/hackage/v/tomland.svg)](https://hackage.haskell.org/package/tomland)
 [![Build status](https://secure.travis-ci.org/kowainik/tomland.svg)](https://travis-ci.org/kowainik/tomland)
-[![Windows build status](https://ci.appveyor.com/api/projects/status/github/kowainik/tomland?branch=master&svg=true)](https://ci.appveyor.com/project/kowainik/tomland)
+[![Hackage](https://img.shields.io/hackage/v/tomland.svg)](https://hackage.haskell.org/package/tomland)
+[![Stackage LTS](http://stackage.org/package/tomland/badge/lts)](http://stackage.org/lts/package/tomland)
+[![Stackage Nightly](http://stackage.org/package/tomland/badge/nightly)](http://stackage.org/nightly/package/tomland)
 [![MPL-2.0 license](https://img.shields.io/badge/license-MPL--2.0-blue.svg)](https://github.com/kowainik/tomland/blob/master/LICENSE)
 
-TOML parser
+> “A library is like an island in the middle of a vast sea of ignorance,
+> particularly if the library is very tall and the surrounding area has been
+> flooded.”
+>
+> ― Lemony Snicket, Horseradish
+
+Bidirectional TOML parser and renderer. See the following blog post with more details:
+
+* TODO: insert link to blog post
+
+This README contains basic usage example of `tomland`. All code below can be
+compiled and run with the following command:
+
+```
+cabal new-build && cabal new-exec readme
+```
+
+## Preamble: imports and language extensions
+
+Since this is a literate haskell file, we need to specify all our language
+extensions and imports up front.
+
+```haskell
+{-# OPTIONS -Wno-unused-top-binds #-}
+
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes       #-}
+
+import Control.Applicative ((<|>))
+import Control.Category ((>>>))
+import Data.Text (Text)
+import Toml (TomlBiMap, TomlCodec, (.=))
+
+import qualified Data.Text.IO as TIO
+import qualified NeatInterpolation as Interpol
+import qualified Toml
+```
+
+## Data type: parsing and printing
+
+Consider the following TOML configuration specified as interpolated text via
+`neat-interpolation` quasi-quoter:
+
+```haskell
+tomlExample :: Text
+tomlExample = [Interpol.text|
```

Comment:
I wonder if it would be useful to have it in the separate toml file, so you won't need to recompile when playing with this example?

Reply:
Hmm.. :thinking: I see value in having this TOML in separate file:

1. Simpler README (no QuasiQuoters and new libraries)
2. Simpler `main` function

Nice idea! Will move this TOML to separate file. Do you think `examples/` directory works as a fine place for this file?

Reply:
Yes, The only issue that I see is it won't be immediately visible to users. Not sure how to solve this, and what is more important

Reply:
I guess it's better to put in separate file. I think users can follow the link and press back button to return :slightly_smiling_face: Not everyone is familiar with quasi-quoters, and `tomland` already uses a lot of non-beginner features... So let's try to make tutorial simpler.